generator/internal/discovery_type_vertex.h

```diff
@@ -70,6 +71,16 @@ class DiscoveryTypeVertex {
   static StatusOr<TypeInfo> DetermineTypeAndSynthesis(
       nlohmann::json const& v, std::string const& field_name);
 
+  struct MessageProperties {
+    std::set<int> reserved_numbers;
+    int max_field_number;
+  };
+
+  // Examines the message Descriptor to determined the reserved field numbers
+  // and the max field number either in use or reserved.
```

Comment:
Q.  What does `max_field_number` buy you?  If 42 was the largest field used, but 12 was unused/unreserved, would we not want to use 12?